codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,439 @@
+    private final StringBuilder sb = new StringBuilder(29); // Sun, 27 Nov 2016 19:37:15 GMT
+    private boolean timeFound;
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private boolean dayOfMonthFound;
+    private int dayOfMonth;
+    private boolean monthFound;
+    private int month;
+    private boolean yearFound;
+    private int year;
+
+    private HttpHeaderDateFormatter() {
+        reset();
+    }
+
+    public void reset() {
+        timeFound = false;
+        hours = -1;
+        minutes = -1;
+        seconds = -1;
+        dayOfMonthFound = false;
+        dayOfMonth = -1;
+        monthFound = false;
+        month = -1;
+        yearFound = false;
+        year = -1;
+        cal.set(Calendar.MILLISECOND, 0);
+        sb.setLength(0);
+    }
+
+    private boolean tryParseTime(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        // h:m:s to hh:mm:ss
+        if (len < 5 || len > 8) {
+            return false;
+        }
+
+        int localHours = -1;
+        int localMinutes = -1;
+        int localSeconds = -1;
+        int currentPartNumber = 0;
+        int currentPartValue = 0;
+
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                currentPartValue = currentPartValue * 10 + getNumericalValue(c);
+            } else if (c == ':') {
+                if (currentPartValue == 0) {
+                    // invalid :: (nothing in between)
+                    return false;
+                }
+                switch (currentPartNumber) {
+                    case 0:
+                        // flushing hours
+                        localHours = currentPartValue;
+                        currentPartValue = 0;
+                        currentPartNumber++;
+                        break;
+                    case 1:
+                        // flushing minutes
+                        localMinutes = currentPartValue;
+                        currentPartValue = 0;
+                        currentPartNumber++;
+                        break;
+                    default:
+                        // invalid, too many :
+                        return false;
+                }
+            } else {
+                // invalid char
+                return false;
+            }
+        }
+
+        if (currentPartValue > 0) {
+            // pending seconds
+            localSeconds = currentPartValue;
+        }
+
+        if (localHours >= 0 && localMinutes >= 0 && localSeconds >= 0) {
+            hours = localHours;
+            minutes = localMinutes;
+            seconds = localSeconds;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len != 1 && len != 2) {
```

Comment:
@slandelle - Understood. You can scope each case as shown in the code above and that should  compile. That is roughly equivalent to the if/else blocks which define a new scope for each condition.

```java
switch(len) {
  case 1: {
     char c0 = ???;
     return ???;
  }
  case 2: {
     char c0 = ???; // c0 is already defined
     char c1 = ???;
     return ???;
  }
}
```